shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -13,49 +51,127 @@ async fn test_hotreload_basic_valkey_connection() {
+    let _: () = con_old
+        .set("persistent_key", "data_from_old_instance")
+        .unwrap();
+    let _: () = con_old.set("counter", 1).unwrap();
+
+    // Verify data is accessible through old instance
+    let stored_value: String = con_old.get("persistent_key").unwrap();
+    assert_eq!(stored_value, "data_from_old_instance");
+
+    // Now start the new instance that will request hot reload
+    let shotover_new = shotover_process("tests/test-configs/hotreload/topology.yaml")
+        .with_hotreload(true)
         .with_log_name("shot_new")
+        .with_hotreload_socket_path("/tmp/shotover-new.sock") // Different socket for new instance
+        .with_hotreload_from_socket(socket_path) // Request handoff from old instance
         .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
         .start()
         .await;
-    let client_b = Client::open("valkey://127.0.0.1:6381").unwrap();
-    let mut con_b = client_b.get_connection().unwrap();
-    let _: () = con_b.set("key_from_b", "value_from_b").unwrap();
 
-    let value_a_from_b: String = con_b.get("key_from_a").unwrap();
-    assert_eq!(value_a_from_b, "value_from_a");
-    let value_b_from_a: String = con_a.get("key_from_b").unwrap();
-    assert_eq!(value_b_from_a, "value_from_b");
+    let client_new = Client::open("valkey://127.0.0.1:6380").unwrap();
+    let mut con_new = client_new
+        .get_connection()
+        .expect("Failed to connect to new instance");
+
+    // Verify that data stored in old instance is still accessible
+    let persistent_value: String = con_new.get("persistent_key").unwrap();
+    assert_eq!(
+        persistent_value, "data_from_old_instance",
+        "Data should persist after hot reload socket handoff"
+    );
+
+    // Test that the old connection continues to function after hot reload
+    assert_valkey_connection_works(
+        &mut con_old,
+        None,
+        &[("persistent_key", "data_from_old_instance")],
+    )
+    .unwrap();
+
+    // Test that old connection can still perform operations
+    let _: () = con_old
+        .set("old_connection_key", "old_still_works")
+        .unwrap();
+
+    // Test increment operation on old connection
+    let old_incr: i32 = con_old.incr("counter", 10).unwrap();
+    assert_eq!(
+        old_incr, 11,
+        "Old connection should handle increment operations after hot reload"
+    );
+
+    // Verify old connection works with counter after hot reload
+    assert_valkey_connection_works(&mut con_old, Some(11), &[]).unwrap();
+
+    // Test setting and getting a new key through old connection
+    let _: () = con_old
+        .set("old_post_reload", "value_set_after_reload")
+        .unwrap();
+
+    // Verify new instance can handle new operations
+    let _: () = con_new.set("new_key", "data_from_new_instance").unwrap();
+
+    // Test that we can increment the counter
+    let counter_value: i32 = con_new.incr("counter", 1).unwrap();
```

Comment:
can this go in `assert_valkey_connection_works` as well? Maybe replace the `connection.get("counter")` logic.